### ISSUE TITLE
Deprecation warning when used with Rails 6

### ISSUE THREAD

**Issue by cgriego** [mongoid/mongoid-rspec#227]
`DEPRECATION WARNING: Ruby 2.5+ (required by Rails 6) provides Hash#transform_values natively, so requiring active_support/core_ext/hash/transform_values is no longer necessary. Requiring it will raise LoadError in Rails 6.1.`

**Comment by jonekdahl** [comment]
I belive this is fixed by #231.

**Comment by rafpiek** [comment]
@jonekdahl There is again the issue when using Rails 6.0.2.2 and Ruby 2.7.1

```
DEPRECATION WARNING: Ruby 2.5+ (required by Rails 6) provides Hash#transform_values natively, so requiring active_support/core_ext/hash/transform_values is no longer necessary. Requiring it will raise LoadError in Rails 6.1. (called from require at /Users/rafalpiekara/Desktop/bs/AG/app/AchieveGuru/backend_achieve-guru/spec/spec_helper.rb:2)
```

I would be glad to take care on that and fix it. I've found the possible code here:

```
require 'active_support/core_ext/hash/transform_values' if Mongoid::Compatibility::Version.mongoid4_or_newer? && ActiveSupport.version < Gem::Version.new('6')
```

It seems to be correct, but the depreaction is present. I've checked the Mongoid repo and there is no Compatibility::Version module. Could it be deprecated? What are your thoughts on that?

**Comment by dblock** [comment]
Mongoid::Compatibility is https://github.com/mongoid/mongoid-compatibility

**Comment by btazi** [comment]
> @jonekdahl There is again the issue when using Rails 6.0.2.2 and Ruby 2.7.1
> 
> ```
> DEPRECATION WARNING: Ruby 2.5+ (required by Rails 6) provides Hash#transform_values natively, so requiring active_support/core_ext/hash/transform_values is no longer necessary. Requiring it will raise LoadError in Rails 6.1. (called from require at /Users/rafalpiekara/Desktop/bs/AG/app/AchieveGuru/backend_achieve-guru/spec/spec_helper.rb:2)
> ```
> 
> I would be glad to take care on that and fix it. I've found the possible code here:
> 
> ```
> require 'active_support/core_ext/hash/transform_values' if Mongoid::Compatibility::Version.mongoid4_or_newer? && ActiveSupport.version < Gem::Version.new('6')
> ```
> 
> It seems to be correct, but the depreaction is present. I've checked the Mongoid repo and there is no Compatibility::Version module. Could it be deprecated? What are your thoughts on that?

I've got the same issue with ruby 2.7.1 rails 6.03 and mongoid 7.1.1, did you manage to solve this ?

**Comment by icortex** [comment]
I'm upgrading my app from Rails 5 to Rails 6.0.3, using Ruby 2.6.5 and mongoid-rspec 4.0.1 and I'm getting the same warning message.

**Comment by lestairon** [comment]
Upgrading my app to Rails 6.0.3 and getting the same warning.

**Comment by dblock** [comment]
Released 4.1.0 that has this resolved.
